internal/controller/operatorconfigmap_controller.go

```diff
@@ -810,84 +796,74 @@ func (c *OperatorConfigMapReconciler) reconcileNoobaaOperatorSubscription() erro
-			return fmt.Errorf("failed to update subscription channel of 'cephcsi-operator' to %v: %v", c.subscriptionChannel, err)
+	if utils.AddLabel(sub, subscriptionLabelKey, subscriptionLabelValue) {
+		if err := c.update(sub); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileRecipeOperatorSubscription() error {
-	recipeOperatorSubscription, err := c.getSubscriptionByPackageName("recipe")
-	if err != nil {
-		return err
-	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != recipeOperatorSubscription.Spec.Channel {
-		recipeOperatorSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(recipeOperatorSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'recipe' to %v: %v", c.subscriptionChannel, err)
+func (c *OperatorConfigMapReconciler) reconcileSubscriptions() error {
+	installPlansForApproval := map[string]struct{}{}
+
+	for _, pkgName := range managedPackageNames {
+		sub, err := c.getSubscriptionByPackageName(pkgName)
+		if client.IgnoreNotFound(err) != nil {
+			return err
 		}
-	}
-	return nil
-}
+		if sub.UID != "" {
+			if c.subscriptionChannel != "" && c.subscriptionChannel != sub.Spec.Channel {
+				sub.Spec.Channel = c.subscriptionChannel
+				if err := c.update(sub); err != nil {
+					return fmt.Errorf("failed to update subscription channel of %s to %v: %v", sub.Name, c.subscriptionChannel, err)
+				}
+			}
 
-func (c *OperatorConfigMapReconciler) reconcileODFSnapshotterSubscription() error {
-	odfSnapshotterSubscription, err := c.getSubscriptionByPackageName("odf-external-snapshotter-operator")
-	if err != nil {
-		return err
-	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != odfSnapshotterSubscription.Spec.Channel {
-		odfSnapshotterSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(odfSnapshotterSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'odf-external-snapshotter-operator' to %v: %v", c.subscriptionChannel, err)
+			ipRef := sub.Status.InstallPlanRef
+			if ipRef != nil {
+				installPlansForApproval[ipRef.Name] = struct{}{}
+			}
 		}
 	}
-	return nil
-}
 
-func (c *OperatorConfigMapReconciler) reconcileWebhookService() error {
-	svc := &corev1.Service{}
-	svc.Name = templates.WebhookServiceName
-	svc.Namespace = c.OperatorNamespace
-	err := c.createOrUpdate(svc, func() error {
-		if err := c.own(svc); err != nil {
+	approvePatch := client.RawPatch(types.MergePatchType, []byte(`{"spec":{"approved":true}}`))
+	for ipName := range installPlansForApproval {
+		installPlan := &opv1a1.InstallPlan{}
+		installPlan.Name = ipName
+		installPlan.Namespace = c.OperatorNamespace
+		if err := c.get(installPlan); client.IgnoreNotFound(err) != nil {
 			return err
+		} else if installPlan.UID != "" && !installPlan.Spec.Approved {
+			if err := c.Patch(c.ctx, installPlan, approvePatch); err != nil {
+				return err
+			}
```

Comment:
Touching an install plane should be done in its own reconcile function. not as part of a reconcile function that deals with changes to subscriptions